src/GetMyCoffee.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+use CoffeeShop\ReadCoffeeShopList;
+use base\Location;
+use CoffeeShop\CoffeeShopsArray;
+
+final class GetMyCoffee{
+    private $__reader;
+    private $__userLocation;
+    private $__coffeeShops;
+    
+    public function __construct($userXCoord, $userYCoord, $coffeeShopsCsv)
+    {
+        $this->readShopsCsv($coffeeShopsCsv);
+        $this->setUserLocationByCoords($userXCoord, $userYCoord);
+        $this->process();
+    }
+
+    public function readShopsCsv($coffeeShopsCsv){
+        $shopsListObj = new ReadCoffeeShopList($coffeeShopsCsv);
+        $this->__reader = $shopsListObj;
+        return $this;
+    }
+
+    public function setUserLocationByCoords($userXCoord, $userYCoord){
+        $userLocation = new Location($userXCoord, $userYCoord);
+        $this->__userLocation = $userLocation;
+        return $this;
+    }
+
+    public function process(){
+        //get the list of shops objs
+        $shopsList = $this->__reader->getList();
+
+        $this->__coffeeShops = new CoffeeShopsArray($shopsList);
+        foreach($this->__coffeeShops as $shop){
+            $shop->calcDistance($this->__userLocation);
```

Comment:
The usage of `CoffeShopDistance` class is really strange, you have to initiate it and then call `calcDistance` for it be 'complete' and usable in the rest of the code.

If you don't know this implementation of this class is very tricky to use it.

Reply:
Yes, indeed this should be well documented.
As a resolution I would extend CoffeeShopsArray into a class that is aware of user location and get that calculation done at the creation time.

Reply:
I feel like are better ideas. Any suggestion?

Reply:
Solved with integrating the user location into the object and calculating the distance into the distance getter.